### ISSUE TITLE
Renderer cannot set selection at the beginning of an attribute element (x<b>{}y</b>)

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5-engine#664]
It thinks that the filler isn't needed in this case, but unfortunately it is. The tests that we have are incorrect and used `domSelection.getRanges()` to check where the selection was rendered, but they should use the anchor and offset which are nicely normalised by Webkit/Blink.

I'll comment some tests out when working on #659, and we'll need to bring them back. It's possible that the only thing that needs to be done in this ticket is to fix `Renderer#_needsInlineFillerAtSelection()`.

**Comment by scofalik** [comment]
This is connected with #629. I already made a PR fixing this issue but it may be overengineered. It all comes to the fact that the selection cannot be placed in `<b>^x</b>` and it is incorrectly (according to view) placed outside of it.

**Comment by scofalik** [comment]
OTOH, this issue concerns only collapsed selection, while the one in #629 is connected with non-collapsed selections too.

We could rewrite `_needAddInlineFiller` so that it takes care of non-collapsed selections too. But that would mean possibiltiy of adding multiple inline fillers.

**Comment by Reinmar** [comment]
I was confused by this ticket for two reasons:

* `anchorNode` is not normalised anymore by Blink and it's possible to place the selection at the beginning of the `<b>` element. In Webkit (Safari) it's still normalised so the selection apparently ends up outside the element. I think that it may have implications for composition when the engine would try to fix the text insertion which happened outside instead of inside the `<b>` element.
* However, besides that, I don't see any other TC where it would have some impact on how the editor works. And I think that it's a rather rare case for someone to press <kbd>Ctrl</kbd>+<kbd>B</kbd> when at `foo{}<b>bar</b>` to start typing with bold at that place.

**Comment by Reinmar** [comment]
cc @f1ames sounds like part of your ZWS changes?

**Comment by f1ames** [comment]
Yes, should be solved by a second inline filler improvement (first is #898 and second _Insert the inline filler when selection is at an inline element boundary_ mentioned in #1342).

**Comment by scofalik** [comment]
"First inline improvement", "Second inline improvement"...

Sounds like some kind of physics laws, like the first law of thermodynamics.
